src/lib/instruction-parser.ts

```diff
@@ -281,9 +359,14 @@ export class InstructionParser {
     const outAccountKey = outAccount.toBase58();
 
     let pointerIndex = swapInstructionIndex + 1;
+
     while (
-      pointerIndex < innerInstructions.length &&
-      !isSwapInstruction(innerInstructions[pointerIndex])
+      this.isNotNextSwap(
```

Comment:
this method name is very confusing. what does it mean?

Reply:
So the search for token transfers must be handled differently for normal swap and custom swap. For normal swap the we can search until we find an instruction whose programId is in AMMs list. 

But custom swaps use two swap program instructions to complete the swap. So in case of custom swap we ignore the second swap program instruction and search for token transfers until we find an instruction whose programId is in AMMs list

I have implemented both of this conditions in a same function `isNotNextSwap` to avoid code repetition. So the function tells when to stop the search based on if the swap is custom or not.